ginivision/src/main/java/net/gini/android/vision/internal/camera/photo/PhotoEdit.java

```diff
@@ -1,88 +1,70 @@
 package net.gini.android.vision.internal.camera.photo;
 
-import android.graphics.Bitmap;
-import android.graphics.BitmapFactory;
 import android.os.AsyncTask;
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 
-import java.io.ByteArrayOutputStream;
+import java.util.ArrayList;
+import java.util.List;
 
 /**
  * @exclude
  */
 public class PhotoEdit {
 
-    private static final int DEF_QUALITY = 100;
-
     private final Photo mPhoto;
-    private int mRotationDegrees;
-    private int mQuality = DEF_QUALITY;
+    private final List<PhotoModifier> mPhotoModifiers;
 
-    public PhotoEdit(@NonNull Photo photo) {
+    PhotoEdit(@NonNull Photo photo) {
         mPhoto = photo;
-        resetToDefaults();
+        mPhotoModifiers = new ArrayList<>();
     }
 
     @NonNull
-    public PhotoEdit rotate(int degrees) {
-        degrees %= 360;
-        mRotationDegrees = degrees;
+    public PhotoEdit rotateTo(int degrees) {
+        final PhotoRotationModifier rotationModifier = new PhotoRotationModifier(degrees, mPhoto);
+        mPhotoModifiers.add(rotationModifier);
         return this;
     }
 
     @NonNull
-    public PhotoEdit compress(int quality) {
-        mQuality = quality;
+    public PhotoEdit compressBy(int quality) {
```

Comment:
Clarified the intent to show that the compression is applied to the any existing compression.